post.js

```diff
@@ -40,42 +40,41 @@ function timeSince (dateString) {
   return Math.floor(seconds) + 's'
 }
 
-function insertImagesAndVideos (content) {
+async function insertImagesAndVideos (content) {
   const parser = new DOMParser()
   const contentDOM = parser.parseFromString(content, 'text/html')
 
-  // Replace all <img> tags with <p2p-image> tags
-  contentDOM.querySelectorAll('img').forEach(img => {
+  const imgPromises = Array.from(contentDOM.querySelectorAll('img')).map(async (img) => {
     const originalSrc = img.getAttribute('src')
-    console.log(`Original img src: ${originalSrc}`)
     const p2pImg = document.createElement('p2p-image')
     p2pImg.setAttribute('src', originalSrc)
+
+    p2pImg.addEventListener('load-error', (event) => {
```

Comment:
Do we need this here? Feels like we could encapsulate this logic entirely in the p2p image component.